terraform/cloud-platform-components/README.md

```diff
@@ -82,6 +83,114 @@ Metrics-server allows us to perform resource queries against the cluster. Comman
+
+```yaml
+    admissionControllerRules:
+      - operations: ["CREATE", "UPDATE"]
+        apiGroups: ["extensions", ""]
+        apiVersions: ["v1beta1”, “v1”]
+        resources: ["ingresses", "namespaces"]
+```
+The admissionControllerRules the operations and resources that the webhook will validate. Intercept API requests when a ingress or a namespace is CREATED or UPDATED, so apiGroups and apiVersions are filled out accordingly (extensions/v1beta1 for ingresses, v1 for namespaces). We can use wildcards (*) for these fields as well.
+
+ ### kube-mgmt
+
+kube-mgmt manages instances of the Open Policy Agent on top of kubernetes. Use kube-mgmt to:
+
+    - Load policies into OPA via kubernetes.
+    - Replicate kubernetes resources including CustomResourceDefinitions (CRDs) into OPA.
+
+ ### OPA-Policies
+
+Policies are written in a high-level declarative language and can be loaded into OPA via the filesystem or well-defined APIs.
+
+kube-mgmt automatically discovers policies stored in ConfigMaps in kubernetes and loads them into OPA. kube-mgmt assumes a ConfigMap contains policies if the ConfigMap is:
+
+    - Created in a namespace listed in the --policies option. Configured in templates/opa/values.yaml.tpl under mgmt/configmapPolicies/namespaces:
+	  - Labelled with openpolicyagent.org/policy=rego.
+
+When a policy has been successfully loaded into OPA, the openpolicyagent.org/policy-status annotation is set to {"status": "ok"}.
+If loading fails for some reason (e.g., because of a parse error), the openpolicyagent.org/policy-status annotation is set to {"status": "error", "error": ...} where the error field contains details about the failure.
+
+IMP NOTE:  Apply a ConfigMap that contains the main OPA policy and default response. This policy is used as an entry-point for policy evaluations and returns allowed:true if policies are not matched to inbound data.
+
+Please see the following documentation by opa on how to do this https://www.openpolicyagent.org/docs/kubernetes-admission-control.html 
+
+ ### How to write Policies
+
+OPA is purpose built for reasoning about information represented in structured documents. The data that our service and its users publish can be inspected and transformed using OPA’s native query language Rego.
+
+Rego was inspired by Datalog, which is a well understood, decades old query language. Rego extends Datalog to support structured document models such as JSON.
+Rego queries are assertions on data stored in OPA. These queries can be used to define policies that enumerate instances of data that violate the expected state of the system.
+
+In Rego, policies are defined inside modules. Modules consist of:
+
+    - Exactly one Package declaration.
+    - Zero or more Import statements.
+    - Zero or more Rule definitions.
+
+Please see the following documentation by opa on how to write Policies https://www.openpolicyagent.org/docs/how-do-i-write-policies.html
+
+ ### kube-mgmt-Replicate
```

Comment:
Should we merge this section into the "kube-mgmt" section above?

I think apart from the configuration example, we can skip the rest and keep it simple.

Reply:
merged this under kube-mgmt

Reply:
Apart from the configuration example, skipped the rest to keep it simple